src/poprox_recommender/components/rankers/__init__.py

```diff
@@ -1,9 +0,0 @@
-from typing import Any
-
-
-class Ranker:
-    def validate_algo_params(self, algo_params: dict[str, Any], valid_params: list[str]):
```

Comment:
We no longer need this base class to share the validation implementation across rankers now that we've made their constructor parameters explicit (instead of a dictionary.)